Add preliminary Freesound provider script

mariuszskon opened the pull request:
## Description
Hello there,
I have implemented some foundational code for extracting data from Freesound. I realise that #282  blocks this. Regardless, I hope the code will be useful at some point. I have based the code off of flickr.py

## Other information
I am interested in participating in the Open Source Internship for [adding audio to the CC Catalog & CC Search](https://opensource.creativecommons.org/internships/project-ideas/all/#add-audio) through Google Summer of Code, so that I can take this PR all the way through to production.

## Checklist
- [x] My pull request has a descriptive title (not a vague title like `Update
  index.md`).
- [x] My pull request targets the `master` branch of the repository.
- [x] My commit messages follow [best practices][best_practices].
- [ ] I added tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no
  visible errors.

[best_practices]:https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

<!-- Make sure you read and understand the following attestation. -->
## Developer Certificate of Origin
```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

mariuszskon committed:
Add preliminary Freesound provider script

mariuszskon committed:
Add AudioStore

mariuszskon committed:
Add tests for AudioStore (naive copy of ImageStore tests)

mariuszskon committed:
Fix AudioStore tests to ensure columns correctly checked

mathemancer reviewed (commented):
Overall, this is perfect (given the incomplete state of the infrastructure to handle the new data.  I noticed in their API docs that there's a 'preview' field.  It's a direct link to a slightly reduced-quality (or range of reduced-quality) version of the sound.  That would probably be good to have in place of the thumbnail I think.

Can't merge until we have the DB schema set up and solidified,  of course.

Overall, great work.  Thank you!

mariuszskon commented on src/cc_catalog_airflow/dags/provider_api_scripts/freesound.py:
```diff
@@ -0,0 +1,220 @@
+
+SECONDS_IN_DAY = 60 * 60 * 24
+FREESOUND_LIMIT = 2000
+DELAY = 1.0  # SECONDS_IN_DAY / FREESOUND_LIMIT
```
Indeed, I semi-accidentally left that in there while testing.

mariuszskon committed:
Add more fields to AudioStore

Add album, file format, duration, genre, language, samplerate, bitdepth, channels.

mariuszskon committed:
Add audio_type field to AudioStore

mariuszskon committed:
[Freesound] Use AudioStore

mariuszskon committed:
[Freesound] Extract description

mariuszskon committed:
[Freesound] Fill as many AudioSource columns as possible with available data

This includes file format, duration, samplerate, bitdepth, channels, and
album.

mariuszskon committed:
[Freesound] Add metadata

mariuszskon committed:
[FreeSound] Add TODO relating to URLs in album and related

mathemancer commented:
Closing until we're ready to work on audio.  Thank you for the work on this, @mariuszskon !
